### ISSUE TITLE
Best way to ensure only new tasks are running?

### ISSUE THREAD

**Issue by jdelman** [silinternational/ecs-deploy#128]
When I use `ecs-deploy`, the new task definition will deploy successfully, but there will sometimes be 1 or 2 containers left over still running the previous task definition. Is there an option I can pass to kill previous version once the deploy is complete?

**Comment by Kostanos** [comment]
In my case draining is happening automatically because of setting on Cluster's service (run only X tasks), but, for most of the services I have the draining may take a lot of time. So yes, would be great to have some settings to reduce the time of draining.

For example:
autoDrainingTimeout: 20s
killOnDrainingnTimeout: true

In this case, script will wait 20 s until auto draining completes, and if it will not be completed, the old tasks will be killed/or not, according to killOnDrainingnTimeout

**Comment by jdelman** [comment]
Hey - I'm not using ECR much anymore, but the actual problem was that it just took a few extra minutes for the old tasks to spin down - not a big deal. I'm going to close this issue.

**Comment by Kostanos** [comment]
why did you closed it? it is still actual for me.
Can we open it and make as feature suggestion?

**Comment by jdelman** [comment]
@Kostanos feel free to re-open it as a new issue

**Comment by Kostanos** [comment]
https://github.com/silinternational/ecs-deploy/issues/147
